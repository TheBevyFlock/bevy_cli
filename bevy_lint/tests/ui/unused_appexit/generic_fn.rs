#![feature(register_tool)]
#![register_tool(bevy)]
//~v NOTE: the lint level is defined here
#![deny(bevy::unused_appexit)]

use bevy::prelude::*;

pub fn foo() {
    App::new().run();
    //~^ ERROR: called `App::run()` without handling the returned `AppExit`
    //~| NOTE: `App::run()` returns `AppExit`, which is used to determine whether the app exited successfully or not
    //~| HELP: consider logging a warning if the returned `AppExit` is an error
    //~| HELP: handle the returned `AppExit`
}
