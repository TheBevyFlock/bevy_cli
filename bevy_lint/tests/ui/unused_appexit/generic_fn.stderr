error: called `App::run()` without handling the returned `AppExit`
 --> tests/ui/unused_appexit/generic_fn.rs:9:16
  |
9 |     App::new().run();
  |     -----------^^^^^
  |     |
  |     help: handle the returned `AppExit`: `let _app_exit = App::new().run()`
  |
  = note: `App::run()` returns `AppExit`, which is used to determine whether the app exited successfully or not
  = help: consider logging a warning if the returned `AppExit` is an error
note: the lint level is defined here
 --> tests/ui/unused_appexit/generic_fn.rs:4:9
  |
4 | #![deny(bevy::unused_appexit)]
  |         ^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

